src/main/java/net/minestom/server/tab/TabListManager.java

```diff
@@ -0,0 +1,96 @@
+package net.minestom.server.tab;
+
+import net.minestom.server.MinecraftServer;
+import net.minestom.server.entity.Player;
+import net.minestom.server.event.player.PlayerDisconnectEvent;
+import net.minestom.server.event.player.PlayerSpawnEvent;
+import net.minestom.server.network.packet.server.play.PlayerInfoPacket;
+import net.minestom.server.tab.populators.DefaultTabPopulator;
+import net.minestom.server.utils.PacketUtils;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.Set;
+import java.util.concurrent.CopyOnWriteArraySet;
+
+public class TabListManager {
+
+    private final Set<TabList> tabLists = new CopyOnWriteArraySet<>();
+    private TabListPopulator tabListPopulator; // needs to be init to default at minimum
+
+    public TabListManager() {
+        this.setTabListPopulator(new DefaultTabPopulator(this));
+        MinecraftServer.getGlobalEventHandler().addEventCallback(PlayerSpawnEvent.class, event -> this.tabListPopulator.onJoin(event.getPlayer()));
+        MinecraftServer.getGlobalEventHandler().addEventCallback(PlayerDisconnectEvent.class, event -> this.tabListPopulator.onLeave(event.getPlayer()));
```

Comment:
I think this might be better to do somewhere else - perhaps in player init? @TheMode

Reply:
I think this would be a good thing to be done inside play init.

Would this therefore be done similarly to setting a player's spawning instance where if it's not set it'll throw an exception?

Reply:
I also believe that the main repo should not rely on events, those are for the library users.

There is a default `TabList`, so you could apply it on login

Reply:
Moved to player init and player#remove